src/sandstorm/hack-session.capnp

```diff
@@ -104,6 +105,15 @@ interface HackSessionContext @0xe14c1f5321159b8f
   revokeApiToken @5 (tokenId :Text);
   # Revoke (delete) a previously-generated token.
 
+  getUIViewForToken @6 (token :Text) -> (session: Grain.UiSession);
+  # Return a UIView corresponding to the token's grain
+
+  getApiEndpoint @7 () -> (endpointUrl :Text);
+  # Return the API's endpoint url
+
+  getExternalWebSession @8 (domain: Text) -> (session :WebSession.WebSession);
```

Comment:
This should perhaps take a WebSession.Params as input, in order to set UserAgent and such.